src/d3a/models/market/market_structures.py

```diff
@@ -252,8 +271,21 @@ def trade_from_JSON_string(trade_string, current_time):
         trade_dict['residual'] = offer_or_bid_from_JSON_string(trade_dict['residual'],
                                                                current_time)
     trade_dict['time'] = parse(trade_dict['time'])
-    # if 'offer_bid_trade_info' in trade_dict:
-    #     trade_dict['offer_bid_trade_info'] = TradeBidInfo(*trade_dict['offer_bid_trade_info'])
+    if key_in_dict_and_not_none(trade_dict, 'offer_bid_trade_info'):
+        if len(trade_dict['offer_bid_trade_info']) == 5:
```

Comment:
```suggestion
        if len(trade_dict['offer_bid_trade_info']) == TradeBidOfferInfo.len():
```
A bit overkill but better to check by actual number of elements in the namedtupple via adding the following implementation inside `class TradeBidOfferInfo`

`    
@classmethod
    def len(cls):
        return len(cls._fields)
`